### PR TITLE
HARP-7218: Use rbush instead of rtree to test screen collisions

### PR DESCRIPTION
**Pull request by robertoraggi**

### CODE REVIEW

File: @here/harp-mapview/lib/ScreenCollisions.ts [hunk 1]
```diff
@@ -72,7 +74,13 @@ export class ScreenCollisions {
      * @param bounds The bounding box in world coordinates.
      */
     allocate(bounds: Math2D.Box): void {
-        this.rtree.insert(bounds, null);
+        const bbox = {
+            minX: bounds.x,
+            minY: bounds.y,
+            maxX: bounds.x + bounds.w,
+            maxY: bounds.y + bounds.h
+        };
```
Comment (robertoraggi) [inline @here/harp-mapview/lib/ScreenCollisions.ts]: it would be nice to remove the need of create bbox objects from Math2D.Box

**Comment by robertoraggi** [comment]
GPU tests please

**Comment by DimitryRd** [comment]
GPU tests please

**Comment by DimitryRd** [comment]
Decoders tests please

**Comment by robertoraggi** [comment]
GPU tests please

File: @here/harp-mapview/lib/ScreenCollisions.ts (hunk 1 above)
Comment (ninok) [reply @here/harp-mapview/lib/ScreenCollisions.ts]: yes, I would also prefer to change the Math2D.Box layout to match the one from rbush.

**Comment by nzjony** [comment]
Local performance tests look good @robertoraggi

**Pull request merged**